lib/sasl.js

```diff
@@ -10,19 +10,34 @@ var debug = require('debug')('amqp10:sasl'),
 
     Connection = require('./connection');
 
+var saslMechanisms = {
+  PLAIN: 'PLAIN',
+  ANONYMOUS: 'ANONYMOUS',
+  NONE: 'NONE'
+};
+
 /**
- * Currently, only supports SASL-PLAIN
+ * Currently, only supports SASL ANONYMOUS or PLAIN
  *
  * @constructor
  */
-function Sasl() {
+function Sasl(mechanism) {
+  if (mechanism && !u.includes(saslMechanisms, mechanism)) {
+    throw new errors.NotImplementedError(
+      'Only SASL PLAIN and ANONYMOUS are supported.');
+  }
+  this.mechanism = mechanism;
   this.receivedHeader = false;
 }
 
+Sasl.Mechanism = saslMechanisms;
```

Comment:
Yeah so I think that was why I left that one plural originally, where we are using it as a list of possible items to find one item in. However, when we export it on the Sasl.Mechanism we are really just using it as a named set of constants that you can choose from. Hence you choose the PLAIN mechanism (singular) constant via Sasl.Mechanism.PLAIN